### PR TITLE
Add sphinx read the docs, organizational docs

### PR DESCRIPTION
**Pull request by AstroBarker**
This PR adds support for sphinx documentation.  It currently contains:
- code of conduct
- governance doc
- contributing
- building
- tracer particles
The code of conduct is also added as markdown. I've also added a couple of badges to the README, for flare.

Unrelated: added `.ccls-cache` to the gitignore.

### CODE REVIEW

File: GOVERNANCE.md
Comment (Yurlungur) [inline GOVERNANCE.md:73]: Given the small number of people currently contributing, I think we should add a clause that the meeting cadence (and strictness of above rules) may depend on project activity.

File: GOVERNANCE.md
Comment (Yurlungur) [inline GOVERNANCE.md:55]: Who's going to start the google mailing list? :)

**Review by Yurlungur — APPROVED** [review]
Two nitpicks below. Also one possible question. Do we want to stick with markdown or switch to sphinx rst and a github pages hosted readthedocs page? The latter is pretty easy to set up.

We can also do both or transition more slowly. We have ZERO docs at the moment, so a RTD page probably doesn't make a lot of sense.

**Comment by AstroBarker** [comment]
> Two nitpicks below. Also one possible question. Do we want to stick with markdown or switch to sphinx rst and a github pages hosted readthedocs page? The latter is pretty easy to set up.
> 
> We can also do both or transition more slowly. We have ZERO docs at the moment, so a RTD page probably doesn't make a lot of sense.

I would love to get started with sphinx RTD, I've just never set it up myself so I don't know the process. Having _something_ available might lessen the barrier to writing docs.

File: GOVERNANCE.md
Comment (AstroBarker) [reply GOVERNANCE.md:73]: I was thinking about this myself when putting it in... I'll add that.

**Comment by Yurlungur** [comment]
> I would love to get started with sphinx RTD, I've just never set it up myself so I don't know the process. Having _something_ available might lessen the barrier to writing docs.

We can copy the setup from parthenon... I can get it set up when I have a spare moment, or I can walk you through it. Main pieces are:
1. Set up a bare branch in the repo to host the generated html
2. Turn on github pages. We'll get the url lanl.github.io/phoebus
3. We can blindly copy the docs github workflow from parthenon, as well as the build script and directory structure
4. Then we add our files and make sure they're listed in the index file at the top level of the source directory of the docs.

**Comment by Yurlungur** [comment]
to be clear, this MR doesn't need to wait on that. We can migrate these into RTD later.

**Comment by Yurlungur** [comment]
Although this would be a good time to do it. ;)

File: CONTRIBUTING.md
Comment (Yurlungur) [inline CONTRIBUTING.md:1]: I have a more detailed "expectations during code review" doc that we can maybe add later. It's in singularity:
https://lanl.github.io/singularity-eos/main/src/contributing.html#expectations-for-code-review

File: CONTRIBUTING.md
Comment (Yurlungur) [inline CONTRIBUTING.md:31]: We have a hard time getting 2 people to look at each MR right now. So I would change this to minimum 1 maintainer, ideally at least 2 people. And specify that you cannot review your own MR.

File: CONTRIBUTING.md
Comment (AstroBarker) [reply CONTRIBUTING.md:1]: I've read this and like it a lot, it's worth threading in I think

**Comment by AstroBarker** [comment]
Yeah, I think this is the right spot to set up a minimal RTD

File: GOVERNANCE.md
Comment (AstroBarker) [reply GOVERNANCE.md:55]: I put it on my todo list. I want a service that isn't google, but probably it will be google..

File: CONTRIBUTING.md
Comment (Yurlungur) [inline CONTRIBUTING.md:29]: Oops sorry one more thing I just noticed. We can do this later, but we should describe the tests process. As well as the github MR checklist. And the style formatting, since we use clang-format. I *think* we can just pull/adapt all of that from parthenon or singularity docs, once we're on sphinx/RTD, so it can maybe be done in a second sweep.

File: CONTRIBUTING.md
Comment (carlnotsagan) [reply CONTRIBUTING.md:1]: +1

File: CONTRIBUTING.md
Comment (carlnotsagan) [inline CONTRIBUTING.md:22]: sign -> adhere to?

File: README.md
Comment (carlnotsagan) [inline README.md:202]: fork? to be consistent with Dev model

**Review by carlnotsagan — APPROVED** [review]

File: CODE_OF_CONDUCT.md
Comment (brryan) [inline CODE_OF_CONDUCT.md:1]: We should not add a code of conduct to phoebus. It's a LANL code, and we already have a code of conduct at LANL that we cannot override. If external participants become a problem relative to LANL standards we can limit their participation appropriately.

Comment (Yurlungur) [reply CODE_OF_CONDUCT.md:1]: If this is a concern, we can instead link to the LANL code of conduct. https://cdn.lanl.gov/files/code-of-conduct-1_312ce.pdf

We could summarize the most relevant part of the code of that document which is on page 36:
```
- Foster a mutually respectful working environment that is free from discrimination,
intimidation, and harassment (including sexual harassment).
- Do not discriminate against others on the basis of any characteristic protected by law
or LANL policy.
- Do not engage in behaviors that create an offensive, hostile, or intimidating work
environment, including shouting, abusive language, threats of violence, the use of
obscenities or other non-verbal expressions of aggression, horseplay such as kicking
co-workers’ chairs or using personal insults, or the use of offensive nicknames or other
derogatory remarks.
- Avoid behavior that a reasonable person would find to be demeaning or humiliating.
```

Comment (Yurlungur) [reply CODE_OF_CONDUCT.md:1]: Alternatively, we could link to [this code of conduct](https://github.com/karlgroves/dontbeadick), which I am a fan. (This is mostly a joke. It doesn't address @brryan 's concern.)

Comment (AstroBarker) [reply CODE_OF_CONDUCT.md:1]: I'm open to pointing to LANL code of conduct if this is a concern.

Comment (brryan) [reply CODE_OF_CONDUCT.md:1]: No concerns from me with replacing this PR's COC with references to the LANL code of conduct instead. @Yurlungur's idea of a relevant summary is a good one.

File: doc/sphinx/src/governance.rst
Comment (brryan) [inline doc/sphinx/src/governance.rst:64]: Sorry to keep picking on this (and dropping in on something that has apparently already been discussed elsewehere) but this would be another code of conduct I would be asked to sign to retain e.g. my merge privileges, and I'm not comfortable with that model for a code I've already been working on for years.

The below are reasonable suggestions, but I would ask that this language be converted from signing an MoU to something like, "We ask that maintainers especially keep the following suggestions in mind to help facilitate an efficient and welcoming collaboration". With the LANL COC as a backdrop and this repository being hosted on the LANL github I think we have robust protection against bad actors.

There's also another issue here: astro funding is intermittent at LANL, and people like me may not always have the time and funding to be regular attendees of monthly meetings. But at the same time I wouldn't want my merge privileges to come and go depending on whether I am currently in active developer mode, I think that adds unnecessary friction. Maybe there is a two-tiered approach here, where contributors actively involved in feature development are encouraged to organize/attend a monthly meeting, and changes to the code broader than development of specific features should be submitted to a broader set of maintainers (for example through tagging more people on PRs, or in a github issue proposing a substantial code change).

Comment (AstroBarker) [reply doc/sphinx/src/governance.rst:64]: I completely agree with everything here. I've already changed the language away from "signing," (although I clearly missed a spot) but I like the suggested language more. I've also added an addendum that "meeting cadence and attendance are subject to activity." I certainly don't want to create any friction or revoke any privileges. I'll iterate on these things today. Thanks for your thoughts and sorry that this came out of nowhere.

Comment (carlnotsagan) [reply doc/sphinx/src/governance.rst:64]: FWIW, this concern was brought up and addressed in the proposal. These docs got out of sync on a few key points. This is one of them. I agree with joining when you can for core developers.

**Comment by AstroBarker** [comment]
When folks have time, I would appreciate another look through this. cc: @brryan @curiousmiah

**Comment by AstroBarker** [comment]
Thanks @brryan! I'm going to merge this today if there are no other immediate concerns (cc: @Yurlungur ) so that this is live when the methods paper hits the arxiv tomorrow. I'll open an issue for further thoughts / planning for docs related things.

**Pull request merged**
